subprojects/gradle/buildchecks/src/main/kotlin/com/avito/android/plugin/build_param_check/incremental_check/IncrementalKaptTask.kt

```diff
@@ -0,0 +1,47 @@
+package com.avito.android.plugin.build_param_check.incremental_check
+
+import com.avito.android.plugin.build_param_check.CheckResult
+import com.avito.android.plugin.build_param_check.CheckTaskWithMode
+import com.avito.kotlin.dsl.getBooleanProperty
+import org.gradle.api.Project
+import org.gradle.api.tasks.TaskAction
+
+internal abstract class IncrementalKaptTask : CheckTaskWithMode() {
+
+    @TaskAction
+    fun check() {
+        val incrementalKaptEnabled = project.getBooleanProperty("kapt.incremental.apt", default = false)
+        if (incrementalKaptEnabled) {
+            checkAnnotationProcessors()
+        }
+    }
+
+    private fun checkAnnotationProcessors() {
+        project.subprojects.forEach { subproject ->
+            if (subproject.hasKotlinKapt && subproject.hasRoomKapt) {
+                mode.get().check(subproject) {
+                    if (RoomIncrementalKaptChecker(subproject).isSupported()) {
```

Comment:
Do we check a single annotation processor?
Should we rename the task to IncrementalRoomKaptCheck?

Reply:
Right now we check only Room. But this task is intended to check any processor if it will become necessary in the feature.

Reply:
Understand